### PR TITLE
Fix setSiblingIndex and some code cleanup

### PR DESCRIPTION
**Pull request by pandamicro**
> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
> - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.

Re: cocos-creator/fireball#2963

Changes proposed in this pull request:
- Fix event order not changed in setSiblingIndex
- Made setSiblingIndex working in JSB

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/utils/base-node.js [hunk 1]
```diff
@@ -1815,14 +1815,16 @@ var BaseNode = cc.Class(/** @lends cc.Node# */{
...
-            if ( !CC_JSB ) {
-                cc.renderer.childrenOrderDirty = true;
-                this._parent._sgNode._reorderChildDirty = true;
-                this._parent._reorderChildDirty = true;
-                this._parent._delaySort();
+                var sibling = siblings[i]._sgNode;
+                // Reset zorder to update their arrival order
+                var zOrder = sibling.getLocalZOrder();
+                if (CC_JSB) {
+                    sibling.setLocalZOrder(zOrder+1);
```
Comment (pandamicro) [inline cocos2d/core/utils/base-node.js:1823]: JSB 中设置同样的 zOrder 无效

File: cocos2d/core/utils/base-node.js [hunk 2]
```diff
@@ -1815,14 +1815,16 @@ var BaseNode = cc.Class(/** @lends cc.Node# */{
...
+                }
+                else {
+                    cc.eventManager._setDirtyForNode(siblings[i]);
+                }
+                sibling.setLocalZOrder(zOrder);
```
Comment (jareguo) [inline cocos2d/core/utils/base-node.js]: 这里多设置了一次 JSB 中的 LocalZOrder

Comment (pandamicro) [inline cocos2d/core/utils/base-node.js]: 是为了把它设置回来，这里的初衷并不是想要改 setLocalZOrder

Comment (jareguo) [inline cocos2d/core/utils/base-node.js]: 那就加个注释吧，不然很像是 bug。
另外就没别的方法吗？

Comment (pandamicro) [inline cocos2d/core/utils/base-node.js]: JSB 中这是最简单有效的办法了，因为没有开放更底层的接口

Comment (pandamicro) [inline cocos2d/core/utils/base-node.js]: Done

File: cocos2d/core/utils/base-node.js [hunk 3]
```diff
@@ -1815,14 +1815,17 @@ var BaseNode = cc.Class(/** @lends cc.Node# */{
             // update rendering scene graph, sort them by arrivalOrder
             var siblings = this._parent._children;
             for (var i = 0, len = siblings.length; i < len; i++) {
-                var sibling = siblings[i];
-                sibling._sgNode.arrivalOrder = i;
```
Comment (jareguo) [inline cocos2d/core/utils/base-node.js:1819]: 不好意思我又来了，能不能在 web 平台上保留原先的 arrivalOrder 更新方式，这样代码更直观，用户看到不同平台的不同调用方式，就知道在 JSB 中 setLocalZOrder 的目的其实是要更新 arrivalOrder 了。而且 setLocalZOrder 里面多余的开销太多了，建议在 JSB 上才这么做。

**Comment by pandamicro** [comment]
@jareguo 已更新

**Pull request merged**